vitessce/wrappers.py

```diff
@@ -328,23 +329,31 @@ def create_cells_json(self):
 
     def create_cell_sets_json(self):
         adata = self.adata
-        cell_sets = CellSets(first_node_name = 'Clusters')
 
-        cell_ids = adata.obs.index.tolist()
-        cluster_ids = adata.obs['CellType'].unique().tolist()
-        cell_cluster_ids = adata.obs['CellType'].values.tolist()
+        cell_set_obs_cols = self.cell_set_obs_cols
 
-        cell_cluster_tuples = list(zip(cell_ids, cell_cluster_ids))
+        cell_sets = CellSets()
 
-        for cluster_id in cluster_ids:
-            cell_set = [
-                str(cell_id)
-                for cell_id, cell_cluster_id in cell_cluster_tuples
-                if cell_cluster_id == cluster_id
-            ]
-            cell_sets.add_node(str(cluster_id), ['Clusters'], cell_set)
+        if cell_set_obs_cols is not None:
 
-        return cell_sets.json
+            for clusters_key in cell_set_obs_cols:
```

Comment:
Maybe rename this variable from `clusters_key` to `tree_name` or something - it took me a second to figure out what was going on.

Reply:
Thanks, I have changed the name and added a comment